backend/src/main/kotlin/ru/nsu/ctf/paketnikback/controller/RegexSearchController.kt

```diff
@@ -0,0 +1,92 @@
+package ru.nsu.ctf.paketnikback.controller
+
+import jakarta.validation.Valid
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+import org.springframework.web.bind.MethodArgumentNotValidException
+
+import io.minio.MinioClient
+import io.minio.errors.ErrorResponseException
+import io.minio.StatObjectArgs
+import io.minio.GetObjectArgs
+
+import io.pkts.Pcap
+
+import ru.nsu.ctf.paketnikback.exception.EntityNotFoundException
+import ru.nsu.ctf.paketnikback.exception.InternalServerErrorException
+import ru.nsu.ctf.paketnikback.domain.dto.RegexSearchRequest
+import ru.nsu.ctf.paketnikback.domain.dto.RegexSearchResponse
+import ru.nsu.ctf.paketnikback.utils.logger
+import ru.nsu.ctf.paketnikback.utils.RegexSearchPcapHandler
+
+@RestController
+@RequestMapping("/search")
+class RegexSearchController(
+    private val minioClient: MinioClient
+) {
+    private val bucketName = "traffic-dumps"
+    private val log = logger()
+    
+    @PostMapping
+    fun search( @RequestBody @Valid request: RegexSearchRequest ): ResponseEntity<RegexSearchResponse> {
+        var result = RegexSearchResponse()
+
+        val filename = request.filename
+        
+        try {
+            val regex = Regex(request.regex)
+        } catch (e: Exception) {
+            throw MethodArgumentNotValidException("ERR: Regex ${request.regex} not valid: ${e.message}")
+        }
+
+        val regex = Regex(request.regex)
```

Comment:
1 - Два раза создается regex
2 - MethodArgumentNotValidException используется для автоматической валидации с @Valid, у тебя в RegexSearchRequest её нет. Выброси просто IllegalArgumentException 
```suggestion
        val regex = try {
            Regex(request.regex)
        } catch (e: Exception) {
            throw IllegalArgumentException("ERR: Invalid regex '${request.regex}': ${e.message}")
        }
```

Reply:
Ок